packages/devtools_app/lib/src/shared/primitives/byte_utils.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+import 'dart:math';
+import 'dart:typed_data';
+
+import 'package:intl/intl.dart';
+
+String? prettyPrintBytes(
+  num? bytes, {
+  int kbFractionDigits = 1,
+  int mbFractionDigits = 1,
+  int gbFractionDigits = 1,
+  bool includeUnit = false,
+  num roundingPoint = 1.0,
+  int maxBytes = 52,
+}) {
+  if (bytes == null) {
+    return null;
+  }
+  // TODO(peterdjlee): Generalize to handle different kbFractionDigits.
```

Comment:
nit: change TODO to your name while moving code as peter is unlikely to resolve the todo.